spinoso-string/src/impls.rs

```diff
@@ -9,6 +9,18 @@ use std::io;
 
 use crate::String;
 
+impl Default for String {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
ditto, why can't we `#[derive(Default)]`?

The default for `EncodedString` should be an empty UTF-8 string.

Reply:
Original reason: Because EncodedString didn't implement default :grimacing: 

However, thinking about it now, I don't think EncodedString should be defining this. If someone is asking for a default from `String`, then shouldn't `String` be making that decision about what the default is? EncodedString is just a wrapper at present, and doesn't contain any real decision about what is "the default" so to speak on anything (e.g. should an Enum have a default value?)